File: frontend/src/types/experience.ts
import { ISOYearMonth } from '@/utils/date';

export type ExperienceType = 'Full-time' | 'Part-time' | 'Internship' | 'Freelance' | 'Contract';

export type Experience = {
  id: string | null;
  title: string;
  organization: string;
  type: ExperienceType;
  location: string | null;
  startDate: ISOYearMonth;
  endDate: ISOYearMonth | null;
  bullets: string[];
};

export function emptyExperience<T extends ExperienceType>(type: T): Experience {
  return {
    id: null,
    title: '',
    organization: '',
    location: null,
    startDate: ISOYearMonth.today(),
    endDate: null,
    bullets: [],
    type,
  };
}

File: backend/app/schemas/experience.py
from enum import Enum
from uuid import UUID, uuid4

from pydantic import Field

from app.schemas.dates import ISOYearMonth
from app.schemas.types import CamelModel


class ExperienceType(Enum):
  FULL_TIME = 'Full-time'
  PART_TIME = 'Part-time'
  INTERNSHIP = 'Internship'
  FREELANCE = 'Freelance'
  CONTRACT = 'Contract'


class Experience(CamelModel):
  id: UUID = Field(default_factory=uuid4)
  title: str
  organization: str
  type: ExperienceType
  location: str | None = None
  start_date: ISOYearMonth
  end_date: ISOYearMonth | None = None

  bullets: list[str] = Field(
    default_factory=list,
    description='List of bullet points describing the experience',
  )


class LLMResponseExperience(CamelModel):
  bullets: list[str] = Field(
    default_factory=list,
    description='List of bullet points describing the experience',
  )
